c++/src/kj/table.c++

```diff
@@ -651,7 +651,7 @@ void BTreeImpl::renumber(uint oldRow, uint newRow, const SearchKey& searchKey) {
     auto& node = tree[pos].parent;
     uint indexInParent = searchKey.search(node);
     pos = node.children[indexInParent];
-    if (node.keys[indexInParent] == oldRow) {
+    if (indexInParent < kj::size(node.keys) && node.keys[indexInParent] == oldRow) {
```

Comment:
Can this be triggered via a synthetic test? Might be useful to at least right an explicit test to avoid a regression in the future. If there's no way to trigger it, then is it actually a bug or just something that needs your commit message as a comment?

Reply:
> Can this be triggered via a synthetic test?

As explained in the commit message, it's not actually possible to trigger bad behavior from this.

I don't think it's necessary to write a comment explaining why we need to avoid accessing an array out-of-bounds.